src/main/java/com/amazonaws/encryptionsdk/kmsv2/AwsKmsMrkAwareMasterKeyProvider.java

```diff
@@ -44,31 +41,29 @@
 
   private final boolean isDiscovery_;
   private final DiscoveryFilter discoveryFilter_;
-  private final String discoveryMrkRegion_;
+  private final Region discoveryMrkRegion_;
 
-  private final KmsMasterKeyProvider.RegionalClientSupplier regionalClientSupplier_;
-  private final String defaultRegion_;
+  private final RegionalClientSupplier regionalClientSupplier_;
+  private final Region defaultRegion_;
 
   public static class Builder implements Cloneable {
-    private String defaultRegion_ = getSdkDefaultRegion();
-    private Optional<KmsMasterKeyProvider.RegionalClientSupplier> regionalClientSupplier_ =
-        Optional.empty();
-    private AWSKMSClientBuilder templateBuilder_ = null;
+    private Region defaultRegion_ = getSdkDefaultRegion();
+
+    private Supplier<KmsClientBuilder> builderSupplier_ = null;
```

Comment:
Why is this a supplier now? Is it just to simplify the logic that was previously necessary with all the cloning?

Reply:
The builders in SDKv2 are neither `Cloneable` nor have accessors for fields stored within them. This prevents us from copying the builders that a user provides to the library. Now, the intended path is they configure a `builderSupplier` which configures the builder in a lambda before providing it to the user. This is also how they would configure the credentials.